WiserTaskScheduler/WiserTaskScheduler/Modules/ServerMonitors/Services/ServerMonitorsService.cs

```diff
@@ -0,0 +1,433 @@
+            var objectService = scope.ServiceProvider.GetRequiredService<IObjectsService>();
+            var stringReplacementsService = scope.ServiceProvider.GetRequiredService<IStringReplacementsService>();
+            var databaseHelpersService = scope.ServiceProvider.GetRequiredService<IDatabaseHelpersService>();
+            var gclSettings = scope.ServiceProvider.GetRequiredService<IOptions<GclSettings>>();
+            var wiserItemsServiceLogger = scope.ServiceProvider.GetRequiredService<ILogger<WiserItemsService>>();
+            var gclCommunicationsServiceLogger = scope.ServiceProvider.GetRequiredService<ILogger<CommunicationsService>>();
+            var dataSelectorsServiceLogger = scope.ServiceProvider.GetRequiredService<ILogger<DataSelectorsService>>();
+
+            var templatesService = new TemplatesService(null, gclSettings, databaseConnection, stringReplacementsService, null, null, null, null, null, null, null, null, null, databaseHelpersService);
+            var dataSelectorsService = new DataSelectorsService(gclSettings, databaseConnection, stringReplacementsService, templatesService, null, null, dataSelectorsServiceLogger, null);
+            var wiserItemsService = new WiserItemsService(databaseConnection, objectService, stringReplacementsService, dataSelectorsService, databaseHelpersService, gclSettings, wiserItemsServiceLogger);
+            var gclCommunicationsService = new CommunicationsService(gclSettings, gclCommunicationsServiceLogger, wiserItemsService, databaseConnection, databaseHelpersService);
+
+            int threshold = monitorItem.Threshold;
+
+            //Check which type of server monitor is used.
+            //Cpu is default Value.
+            switch (monitorItem.ServerMonitorType)
+            {
+                case ServerMonitorTypes.Drive:
+                    if (monitorItem.DriveName == "All")
+                    {
+                        await GetAllHardDrivesSpaceAsync(monitorItem, threshold, gclCommunicationsService, configurationServiceName);
+                    }
+                    else
+                    {
+                        await GetHardDriveSpaceAsync(monitorItem, threshold, gclCommunicationsService, configurationServiceName, monitorItem.DriveName);
+                    }
+                    break;
+                case ServerMonitorTypes.Ram:
+                    await GetRAMSpaceAsync(monitorItem, threshold, gclCommunicationsService, configurationServiceName);
+                    break;
+                case ServerMonitorTypes.Cpu:
+                    await GetCpuUsageAsync(monitorItem, threshold, gclCommunicationsService, configurationServiceName);
+                    break;
+                case ServerMonitorTypes.Network:
+                    await GetNetworkUtilization(monitorItem, threshold, gclCommunicationsService, configurationServiceName);
+                    break;
+                default:
+                    break;
```

Comment:
Deze `default` niet leeg laten. In zulke gevallen altijd een `throw new ArgumentOutOfRangeException` toevoegen, zodat we een duidelijke foutmelding krijgen wanneer iemand een optie gebruikt waar we nog niets mee doen. Dan is het in de toekomst heel makkelijk om te zien als ooit een nieuwe optie toevoegen aan de enum en die dan hier vergeten af te handelen.